pkg/draft/pack/archive.go

```diff
@@ -0,0 +1,171 @@
+package pack
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+
+	"github.com/ghodss/yaml"
+
+	"k8s.io/helm/pkg/proto/hapi/chart"
+)
+
+// Archive creates an archived pack to the given directory.
+//
+// This takes an existing pack and a destination directory.
+//
+// If the directory is /foo, and the pack is named bar, this
+// will generate /foo/bar.tgz.
+//
+// This returns the absolute path to the pack archive file.
+func Archive(p *Pack, outDir string) (string, error) {
+
+	// Create archive
+	if fi, err := os.Stat(outDir); err != nil {
+		return "", err
+	} else if !fi.IsDir() {
+		return "", fmt.Errorf("location %s is not a directory", outDir)
+	}
+
+	name := p.Metadata.Name
+	if name == "" {
+		return "", errors.New("no pack name specified")
+	}
+
+	filename := fmt.Sprintf("%s.tgz", name)
+	filename = filepath.Join(outDir, filename)
+	f, err := os.Create(filename)
+	if err != nil {
+		return "", err
+	}
+
+	// Wrap in gzip writer
+	zipper := gzip.NewWriter(f)
+	// zipper.Head.Extra = headerBytes // Helm uses this. Why?
```

Comment:
@technosophos @adamreese do you know why or know anyone that knows why we do this for Helm?